### ISSUE TITLE
ppc64le all-in-one image is missing kubectl

### ISSUE THREAD

**Issue by Rajalakshmi-Girish** [ppc64le-cloud/test-infra#133]
The latest version of the all-in-one image available in quay.io/powercloud is **missing kubectl when used on ppc64le architecture.**
The kubectl is available when installed on x86. Below are the observations on the pods created with the all-in-one image on x86 and ppc64le:

**x86_64:**
```
root@ragini-x861:~/raji# cat test-kubectl-pod.yaml
apiVersion: v1
kind: Pod
metadata:
 name: test-kubectl-pod
spec:
 containers:
   - image: quay.io/powercloud/all-in-one:0.1
     name: test-container
     command: [ "sleep" ]
     args: [ "infinity" ]
root@ragini-x861:~/raji# uname -p
x86_64
root@ragini-x861:~/raji# kubectl exec -it test-kubectl-pod /bin/bash
kubectl exec [POD] [COMMAND] is DEPRECATED and will be removed in a future version. Use kubectl exec [POD] -- [COMMAND] instead.
root@test-kubectl-pod:/workspace# kubectl version
Client Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.13-dispatcher", GitCommit:"fd22db44e150011eccc8729db223945384460143", GitTreeState:"clean", BuildDate:"2020-07-24T07:27:52Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-13T13:20:00Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
root@test-kubectl-pod:/workspace#

```
**on ppc64le:**
```
[root@rajalakshmi-workspace ~]# cat test-kubectl-pod.yaml
apiVersion: v1
kind: Pod
metadata:
 name: test-kubectl-pod
spec:
 containers:
   - image: quay.io/powercloud/all-in-one:0.1
     name: test-container
     command: [ "sleep" ]
     args: [ "infinity" ]
[root@rajalakshmi-workspace ~]# uname -p
ppc64le
[root@rajalakshmi-workspace ~]# kubectl exec -it test-kubectl-pod /bin/bash
kubectl exec [POD] [COMMAND] is DEPRECATED and will be removed in a future version. Use kubectl exec [POD] -- [COMMAND] instead.
root@test-kubectl-pod:/workspace# kubectl version
bash: kubectl: command not found
root@test-kubectl-pod:/workspace#
```

**Comment by Rajalakshmi-Girish** [comment]
@mkumatag @RagniGarg97 ^^

**Comment by mkumatag** [comment]
I think kubectl is getting installed part of a some Google tools, let's override the existing tool and install new one

**Comment by RagniGarg97** [comment]
**on ppc64le:**

while using gcloud components install kubectl  , the Path is not getting set and also I cannot find the binaries of kubectl
```
Installation Root: [/google-cloud-sdk]
Installed Components:
  kubectl: [1.15.11]
  beta: [2020.09.03]
  gsutil: [4.53]
  alpha: [2020.09.03]
  core: [2020.09.03]
System PATH: [/google-cloud-sdk/bin:/workspace:/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin]
Python PATH: [/google-cloud-sdk/lib/third_party:/google-cloud-sdk/lib:/usr/lib/python37.zip:/usr/lib/python3.7:/usr/lib/python3.7/lib-dynload]
Cloud SDK on PATH: [True]
Kubectl on PATH: [False]
```
also `.kube` folder is not getting generated

but if we do gcloud components list it shows that kubectl is installed
```
# gcloud components list

Your current Cloud SDK version is: 327.0.0
The latest available version is: 327.0.0

┌──────────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                              Components                                              │
├───────────────┬────────────────────────────────────────────────┬──────────────────────────┬──────────┤
│     Status    │                      Name                      │            ID            │   Size   │
├───────────────┼────────────────────────────────────────────────┼──────────────────────────┼──────────┤
│ Not Installed │ BigQuery Command Line Tool                     │ bq                       │  < 1 MiB │
│ Not Installed │ Cloud Datalab Command Line Tool                │ datalab                  │  < 1 MiB │
│ Not Installed │ Cloud Datastore Emulator                       │ cloud-datastore-emulator │ 18.4 MiB │
│ Not Installed │ Cloud Firestore Emulator                       │ cloud-firestore-emulator │ 41.6 MiB │
│ Not Installed │ Cloud Pub/Sub Emulator                         │ pubsub-emulator          │ 56.5 MiB │
│ Not Installed │ gcloud app Java Extensions                     │ app-engine-java          │ 58.1 MiB │
│ Not Installed │ gcloud app Python Extensions                   │ app-engine-python        │  6.1 MiB │
│ Not Installed │ gcloud app Python Extensions (Extra Libraries) │ app-engine-python-extras │ 27.1 MiB │
│ Not Installed │ pkg                                            │ pkg                      │          │
│ Installed     │ Cloud SDK Core Libraries                       │ core                     │ 16.8 MiB │
│ Installed     │ Cloud Storage Command Line Tool                │ gsutil                   │  3.9 MiB │
│ Installed     │ gcloud Alpha Commands                          │ alpha                    │  < 1 MiB │
│ Installed     │ gcloud Beta Commands                           │ beta                     │  < 1 MiB │
│ Installed     │ kubectl                                        │ kubectl                  │  < 1 MiB │
└───────────────┴────────────────────────────────────────────────┴──────────────────────────┴──────────┘
To install or remove components at your current SDK version [327.0.0], run:
  $ gcloud components install COMPONENT_ID
  $ gcloud components remove COMPONENT_ID

To update your SDK installation to the latest version [327.0.0], run:
  $ gcloud components update
```

**Comment by Rajalakshmi-Girish** [comment]
The above output og gcloud component list shows a kubectl which is less than 1MB and this is not the actual one!
While I build the image in x8 I see  ---> 1343c42f8b24
Step 7/7 : RUN wget -q https://dl.google.com/dl/cloudsdk/channels/rapid/google-cloud-sdk.tar.gz &&     tar xzf google-cloud-sdk.tar.gz -C / &&     rm google-cloud-sdk.tar.gz &&     gcloud components install alpha beta kubectl gsutil &&     gcloud info | tee /workspace/gcloud-info.txt
 ---> Running in d2f6f7ee5acb


Your current Cloud SDK version is: 327.0.0
Installing components from version: 327.0.0

+-----------------------------------------------------------------------------+
|                     These components will be installed.                     |
+-----------------------------------------------------+------------+----------+
|                         Name                        |  Version   |   Size   |
+-----------------------------------------------------+------------+----------+
| Cloud SDK Core Libraries (Platform Specific)        | 2021.02.05 |  < 1 MiB |
| Cloud Storage Command Line Tool                     |       4.58 |  3.9 MiB |
| Cloud Storage Command Line Tool (Platform Specific) |       4.58 |  < 1 MiB |
| gcloud Alpha Commands                               | 2019.05.17 |  < 1 MiB |
| gcloud Beta Commands                                | 2019.05.17 |  < 1 MiB |
| gcloud cli dependencies                             | 2020.06.12 |  < 1 MiB |
| kubectl                                             |    1.17.17 |  < 1 MiB |
| kubectl                                             |    1.17.17 | 73.9 MiB |
+-----------------------------------------------------+------------+----------+

@mkumatag shall we just override the existing tool OR must look more into why `gcloud component install kubectl` is only working on amd and not in ppc

**Comment by mkumatag** [comment]
> @mkumatag shall we just override the existing tool OR must look more into why `gcloud component install kubectl` is only working on amd and not in ppc

Usually, even if it gets installed via gcloud, kubectl won't be that latest. We should look at installing the latest version from the k8s release

**Comment by RagniGarg97** [comment]
https://github.com/ppc64le-cloud/test-infra/pull/134

**Comment by mkumatag** [comment]
fixed via #134, hence closing this task
